rgen/rgen-placer/src/placer/tree/better_bushes.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{Placer, Random, Result, Rng, rng};
+use rgen_base::{BlockFilter, BlockState, Pos};
+use rgen_world::{PartialWorld, UndoError};
+use std::collections::HashMap;
+
+pub struct BetterBush {
+  pub place_above:   BlockFilter,
+  pub trunk:         BlockState,
+  pub leaves:        BlockState,
+  pub avg_per_chunk: f64,
+}
+
+impl Default for BetterBush {
+  fn default() -> Self {
+    Self {
+      place_above:   block![grass].into(),
+      trunk:         block!(log[variant = "jungle"]),
+      leaves:        block![leaves[variant = "jungle", check_decay = false, decayable = true]],
+      avg_per_chunk: 30.0, //12
+    }
+  }
+}
+
+impl Placer for BetterBush {
+  fn radius(&self) -> u8 { 10 }
+
+  fn avg_per_chunk(&self) -> f64 { self.avg_per_chunk }
+
+  fn place(&self, world: &mut PartialWorld, rng: &mut Rng, pos: Pos) -> Result {
+    if pos.y + 24 >= 255 || pos.y <= 1 {
+      return Err(UndoError);
+    }
+
+    let below = pos + Pos::new(0, -1, 0);
+    if !self.place_above.contains(world.get(below)) || world.get(pos) != block![air] {
+      return Err(UndoError);
+    }
+    if self.can_place_bush(world, pos) {
+      self.bush_placer(world, rng, pos);
+    }
+    Ok(())
+  }
+}
+
+impl BetterBush {
+  fn can_place_bush(&self, world: &PartialWorld, base: Pos) -> bool {
+    for dy in 0..2 {
+      for dz in -1..=1 {
+        for dx in -1..=1 {
+          let pos = base + Pos::new(dx, dy, dz);
+          let block = world.get(pos);
+          if block != block![air] && block != self.leaves {
+            return false;
+          }
+        }
+      }
+    }
+    true
+  }
+  fn bush_placer(&self, world: &mut PartialWorld, rng: &mut Rng, base: Pos) {
+    let bottoms = vec![
+      vec!["-ll--", "-lll-", "-lcll", "-lll-", "-----"], // uses top 1
+      vec!["--ll-", "-llll", "-lcll", "-lll-", "-----"], // uses top 2
+      vec!["--l--", "-llll", "-lcll", "-lll-", "-----"], // uses top 1
+      vec!["-----", "-lll-", "-lcl-", "-lll-", "-----"], // uses top 3
+      vec!["-ll--", "-ll--", "-lcl-", "--ll-", "-----"], // uses top 1
+      vec!["--l--", "-lll-", "-lcl-", "-ll--", "-----"], // uses top 4
+    ];
+
+    let tops = vec![
+      vec!["-l-", "lcl", "-l-"], // top 1
+      vec!["-ll", "lcl", "-l-"], // top 2
+      vec!["-l-", "lcl", "ll-"], // top 3
+      vec!["-l-", "lc-", "-l-"], // top 4
+    ];
```

Comment:
use arrays, not `vec!`. no need to make a vector for these, as they're constant.